nautobot/users/views.py

```diff
@@ -213,6 +226,130 @@ def post(self, request):
         )
 
 
+#
+# Saved Views
+#
+
+
+class SavedViewUIViewSet(
+    ObjectDetailViewMixin,
+    ObjectBulkDestroyViewMixin,
+    ObjectChangeLogViewMixin,
+    ObjectDestroyViewMixin,
+    ObjectEditViewMixin,
+    ObjectListViewMixin,
+):
+    queryset = SavedView.objects.all()
+    filterset_class = SavedViewFilterSet
+    table_class = SavedViewTable
+    action_buttons = ("export",)
+
+    def get_table_class_string_from_view_name(self, view_name):
+        app_label, model_name = view_name.split(":")
+        model_name = model_name.replace("_list", "")
+        model = apps.get_model(app_label=app_label, model_name=model_name)
+        return f"{model.__name__}Table"
+
+    def retrieve(self, request, *args, **kwargs):
+        """
+        The detail view for a saved view should the related ObjectListView
+        with saved configurations applied
+        """
+        instance = self.get_object()
+        query_string = instance.view_config
+        list_view_url = reverse(instance.view) + query_string + f"&saved_view={instance.pk}"
+        return redirect(list_view_url)
+
+    def update(self, request, *args, **kwargs):
+        """
+        request.GET: render the ObjectEditForm which is passed to NautobotHTMLRenderer as Response.
+        request.POST: call perform_update() which validates the form and perform the action of update/partial_update of an existing object.
+        Override to add more variables to Response.
+        """
+        sv = self.get_queryset().get(pk=request.GET.get("saved_view", None))
+        table_changes_pending = request.GET.get("table_changes_pending", False)
+        pagination_count = request.GET.get("per_page", None)
+        if pagination_count is not None:
+            sv.pagination_count = int(pagination_count)
+        sort_order = request.GET.getlist("sort", [])
+        sv.sort_order = sort_order
+
+        new_filter_params = []
+        for key, _ in request.GET.items():
+            if key in self.non_filter_params:
+                continue
+            sv.filter_params[key] = request.GET.getlist(key)
+            new_filter_params.append(key)
+
+        # delete filter params that are no longer in use
+        for param in list(sv.filter_params.keys()):
+            if param not in new_filter_params:
+                sv.filter_params.pop(param)
+        if table_changes_pending:
+            table_class = self.get_table_class_string_from_view_name(sv.view)
+            sv.table_config[f"{table_class}"] = request.user.get_config(f"tables.{table_class}")
+        sv.validated_save()
+        query_string = sv.view_config
+        list_view_url = reverse(sv.view) + query_string + f"&saved_view={sv.pk}"
+        messages.success(request, f"Successfully updated current view {sv.name}")
+        return redirect(list_view_url)
```

Comment:
Why do we need to include `query_string` in the redirect? Shouldn't just `?saved_view={sv.pk}` be sufficient to render the updated view?